### PR TITLE
[decoupled-execution] Integration Part 1: Bridging BufferManager and EpochManager

### PR DESCRIPTION
**Pull request by yuxiamit**
+ Bridging BufferManager and EpochManager
+ Clean tests

I will make buffer manager sync in the next PR.

## Motivation

(Write your motivation for proposed changes here.)

### Have you read the [Contributing Guidelines on pull requests](https://github.com/diem/diem/blob/main/CONTRIBUTING.md#pull-requests)?

(Write your answer here.)

## Test Plan

(Share your test plan here. If you changed code, please provide us with clear instructions for verifying that your changes work.)

## Related PRs

(If this PR adds or changes functionality, please take some time to update the docs at https://github.com/diem/diem/tree/main/developers.diem.com, and link to your PR here.)

## If targeting a release branch, please fill the below out as well

 * Justification and breaking nature (who does it affect? validators, full nodes, tooling, operators, AOS, etc.)
 * Comprehensive test results that demonstrate the fix working and not breaking existing workflows.
 * Why we must have it for V1 launch.
 * What workarounds and alternative we have if we do not push the PR.

### CODE REVIEW

File: consensus/src/experimental/buffer_manager.rs [hunk 1]
```diff
@@ -56,6 +57,10 @@ pub type BufferItemRootType = Link<BufferItem>;
 pub type Sender<T> = UnboundedSender<T>;
 pub type Receiver<T> = UnboundedReceiver<T>;
 
+pub fn create_channel<T>() -> (Sender<T>, Receiver<T>) {
```
[the remark this answers is not in the document]
Comment (yuxiamit) [reply consensus/src/experimental/buffer_manager.rs:60]: I want to wrap the choice of channels inside this file for better modularity.

File: consensus/src/epoch_manager.rs [hunk 2]
```diff
@@ -551,15 +643,25 @@ impl EpochManager {
                     "process_sync_info",
                     p.process_sync_info_msg(*sync_info, peer_id).await
                 ),
-                verified_event @ VerifiedEvent::CommitVote(_)
-                | verified_event @ VerifiedEvent::CommitDecision(_) => {
+                VerifiedEvent::CommitVote(commit_vote) => {
+                    if let Some(sender) = &mut self.commit_msg_tx {
+                        sender
+                            .push(commit_vote.author(), VerifiedEvent::CommitVote(commit_vote))
```
[the remark this answers is not in the document]
Comment (yuxiamit) [reply consensus/src/epoch_manager.rs]: resolved

File: consensus/src/epoch_manager.rs [hunk 3]
```diff
@@ -551,15 +643,25 @@ impl EpochManager {
...
+                }
+                VerifiedEvent::CommitDecision(commit_decision) => {
                     if let Some(sender) = &mut self.commit_msg_tx {
-                        sender.send(verified_event).await.map_err(|err| {
-                            anyhow!(
-                                "Error in Passing Commit Message (CommitVote/CommitDecision): {}",
-                                err
+                        sender
+                            .push(
+                                commit_decision.author(),
```
[the remark this answers is not in the document]
Comment (yuxiamit) [reply consensus/src/epoch_manager.rs]: resolved

**Review by zekun000 — DISMISSED** [review]

**Pull request closed without merging**